sky/packages/sky/lib/src/rendering/stack.dart

```diff
@@ -237,3 +270,40 @@ class RenderStack extends RenderBox with ContainerRenderObjectMixin<RenderBox, S
     }
   }
 }
+
+/// Implements the same layout algorithm as RenderStack but only paints the child
+/// specified by index.
+class RenderIndexedStack extends RenderStack {
+  RenderIndexedStack({
+    List<RenderBox> children,
+    double horizontalAlignment: 0.0,
+    double verticalAlignment: 0.0,
+    int index: 0
+  }) : _index = index, super(
+   children: children,
+   horizontalAlignment: horizontalAlignment,
+   verticalAlignment: verticalAlignment
+  );
+
+  int get index => _index;
+  int _index;
+  void set index (int value) {
+    if (_index != value) {
+      _index = value;
+      markNeedsLayout();
+    }
+  }
+
+  void defaultPaint(PaintingContext context, Offset offset) {
```

Comment:
overriding defaultPaint() is a bit weird, since by definition then it's not the default any more...